test/test_cpp_extensions_aot.py

```diff
@@ -0,0 +1,162 @@
+import os
+import shutil
+import sys
+import unittest
+import warnings
+
+import torch.testing._internal.common_utils as common
+from torch.testing._internal.common_utils import IS_WINDOWS
+from torch.testing._internal.common_cuda import TEST_CUDA
+import torch
+import torch.backends.cudnn
+import torch.utils.cpp_extension
+
+try:
+    import torch_test_cpp_extension.cpp as cpp_extension
+    import torch_test_cpp_extension.msnpu as msnpu_extension
+except ImportError:
+    warnings.warn(
+        "test_cpp_extensions_aot.py cannot be invoked directly. Run "
+        "`python run_test.py -i test_cpp_extensions_aot` instead."
+    )
+
+
+class TestCppExtensionAOT(common.TestCase):
+    """Tests ahead-of-time cpp extensions
+
+    NOTE: run_test.py's test_cpp_extensions_aot_no_ninja target
+    also runs this test case, but with ninja disabled. If you are debugging
+    a test failure here from the CI, check the logs for which target
+    (test_cpp_extensions_aot vs test_cpp_extensions_aot_no_ninja)
+    failed.
+    """
+
+    def setUp(self):
+        default_build_root = torch.utils.cpp_extension.get_default_build_root()
+        if os.path.exists(default_build_root):
+            shutil.rmtree(default_build_root)
```

Comment:
I know you didn't write this code, but this looks soooooo sketchy

Reply:
It turns out we don't actually need that here so I deleted it. It still exists in test_cpp_extensions_jit (this code is used for wiping the build folder for just-in-time cpp extensions)